license.md

```diff
@@ -60,9 +60,9 @@ You need not contribute prototype changes, extensions, or applications that you
 
 ## Applications
 
-The purpose of this exception to [Copyleft](#copyleft) is to encourage cooperative development of this software, and to prohibit use of this software to create competing substitutes for this software that are not contributed.
+The purpose of this exception to [Copyleft](#copyleft) is to encourage cooperative development of this software by prohibiting use of this software to create competing, proprietary substitutes.
 
-You need not contribute software that only invokes this software's functionality through the interfaces this software exposes, without exposing this software's interfaces and functionality to other software.
+You need not contribute software that only invokes this software's functionality through the interfaces this software exposes, without exposing this software's interfaces or functionality to other software.  Exposing this software's functionality only in specialized application to a particular use case does not count as exposing this software's functionality.
```

Comment:
Hmmmm, what if you make, say, database software, and someone makes a thin wrapper that pre-processes strings in a way only useful in bioinformatics (or whatever), and then sells that as a "bioinformatics database as a service"? That seems like it would qualify as a "specialized application to a particular use case". Would that go against your intent? Or, as long as it doesn't work as a general-purpose substitute for your database, you'd be A-OK with that?

Reply:
I would _not_ read the terms above to require contributing that bioinformatics database.

Reply:
I wouldn't either; is that your intent in crafting this license? Even though it's a thin wrapper that offers almost entirely the same functionality, because it's specialized for bioinformatics, it wouldn't need to be open-sourced?

Reply:
Ref: https://github.com/kemitchell/api-copyleft-license/issues/35#issuecomment-499280260